### ISSUE TITLE
Build failing on site with blog - 8.5.5-insiders-4.24.2 

### ISSUE THREAD

**Issue by Techbrunch** [squidfunk/mkdocs-material#4438]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When building with 8.5.5-insiders-4.24.2 the CI is broken.

### Expected behaviour

The CI build without errors.

### Actual behaviour

```
Run mkdocs gh-deploy --force --config-file mkdocs.yml
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.7/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/__main__.py", line 209, in gh_deploy_command
    cfg = config.load_config(
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/base.py", line 224, in load_config
    errors, warnings = cfg.validate()
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/base.py", line 108, in validate
    run_failed, run_warnings = self._validate()
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/base.py", line 63, in _validate
    self[key] = config_option.validate(value)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 132, in validate
    return self.run_validation(value)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 690, in run_validation
    plgins[item] = self.load_plugin(item, cfg)
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 703, in load_plugin
    Plugin = self.installed_plugins[name].load()
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/importlib_metadata/__init__.py", line 207, in load
    module = import_module(match.group('module'))
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/material/plugins/search/plugin.py", line 42, in <module>
    class SearchPluginConfig(BasePlugin.config_class):
AttributeError: type object 'SearchPlugin' has no attribute 'config_class'
```

### Steps to reproduce

I just pushed a new blog post and when using the previous tag it works.

### Package versions

- Python: `3.10.7`
- MkDocs: `mkdocs-1.4.0`
- Material: `mkdocs-material-8.5.5+insiders.4.24.2`

### Configuration

```yaml
site_name: Today I Learned
plugins:
  - search
  - social
  - privacy
  - tags
  - git-revision-date-localized:
      enable_creation_date: true
      fallback_to_build_date: true
  - git-authors
  - blog:
      post_url_format: "{slug}"
  - rss:
      match_path: blog/posts/.*
      date_from_meta:
        as_creation: date
      categories:
        - categories
        - tags
...
```

### System information

CI running in a GitHub action:

```
name: ci 
on: push
jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: 3.x
      - uses: actions/cache@v2
        with:
          key: ${{ github.ref }}
          path: .cache
      - run: sudo apt-get install libcairo2-dev libfreetype6-dev libffi-dev libjpeg-dev libpng-dev libz-dev
      - run: pip install git+https://xxx@github.com/squidfunk/mkdocs-material-insiders.git
      - run: pip install mkdocs-git-revision-date-localized-plugin
      - run: pip install mkdocs-git-committers-plugin-2
      - run: pip install mkdocs-git-authors-plugin
      - run: pip install pillow cairosvg
      - run: pip install mkdocs-rss-plugin
      - run: mkdocs gh-deploy --force --config-file mkdocs.yml
```

**Comment by squidfunk** [comment]
Thank for reporting. The error should already be fixed, as I removed all dependencies to the MkDocs search plugin in https://github.com/squidfunk/mkdocs-material-insiders/commit/b539b5dbaf7bc1442aed6674f2944901221307a8. Please try to rebuild your documentation and check whether the issue persists. [I just did the same, and the builds succeeded](https://github.com/squidfunk/mkdocs-material/actions/runs/3168415860).

**Comment by oprypin** [comment]
mkdocs-material requires MkDocs 1.4 but you somehow upgraded it bypassing that requirement because your MkDocs is evidently not 1.4.

>       File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 690, in run_validation
>         plgins[item] = self.load_plugin(item, cfg)

The code line just doesn't match up:

https://github.com/mkdocs/mkdocs/blob/1.4.0/mkdocs/config/config_options.py#L690

**Comment by Techbrunch** [comment]
I suspect there is some caching on the GitHub action side. I did try to remove `actions/cache@v2` but this did not change the result.

**Comment by oprypin** [comment]
Try changing all `pip install` to `pip install -U`

**Comment by Techbrunch** [comment]
I found the culprit:

```
mkdocs-rss-plugin 1.1.0 requires mkdocs<1.4,>=1.1, but you have mkdocs 1.4.0 which is incompatible.
```

I disabled the RSS plugin and now my build is passing \o/

**Comment by squidfunk** [comment]
Thanks @oprypin for investigating! Closing as upstream issue.

**Comment by squidfunk** [comment]
The problems with the `mkdocs-rss-plugin` should be fixed in squidfunk/mkdocs-material-insiders@16a4cf8
 and were released as [8.5.6+insiders-4.25.1](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/8.5.6-insiders-4.25.1).
